### PR TITLE
Setup VPA and use goldilocks to autogenerate VPA requests

### PR DESCRIPTION
**Pull request by NissesSenap**
For us to be able to give simple suggestions to our customers we should install VPA in our clusters and enable it in recommendation mode. This way we can easily provide an overview of this data in our reports and it will also be simple to use for our developers.

Partly solves: https://github.com/XenitAB/terraform-modules/issues/505

### CODE REVIEW

File: modules/kubernetes/vpa/templates/vpa-values.yaml.tpl [hunk 1]
```diff
@@ -0,0 +1,36 @@
+priorityClassName: platform-medium
+
+recommender:
+  # recommender.enabled -- If true, the vpa recommender component will be installed.
+  enabled: true
+  # recommender.extraArgs -- A set of key-value flags to be passed to the recommender
+  extraArgs:
+    v: "4"
```
Comment (NissesSenap) [inline modules/kubernetes/vpa/templates/vpa-values.yaml.tpl:8]: Should lower this logging, VPA generates lots of logs

File: modules/kubernetes/vpa/templates/vpa-values.yaml.tpl [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    pod-recommendation-min-cpu-millicores: 15
+    pod-recommendation-min-memory-mb: 24
+    prometheus-address: http://prometheus-operated.prometheus.svc.cluster.local:9090
+    storage: prometheus
+    # How much time back prometheus have to be queried to get historical metrics
+    # history-length: 8
```
Comment (NissesSenap) [inline modules/kubernetes/vpa/templates/vpa-values.yaml.tpl:14]: Got `Could not initialize history provider: history length 8 is not a valid Prometheus duration: not a valid duration string: 8`
Probably need to define if it should be hours days or something like that. Need to read the docs. 8 days is the default value

**Comment by NissesSenap** [comment]
* Currently haven't added if in kube-state-metrics to add VPA.
* Haven't added this feature to EKS.
* Should add podmonitor config for prometheus of the VPA.

**Review by jimgus — COMMENTED** [review]
I think it can be good to mention somewhere in the Readme that we only use the recommend part of VPA

**Comment by NissesSenap** [comment]
The current plan is to use goldilocks to auto create VPA config.
There are two ways to go, ether we add two flags:

```
--on-by-default
--exclude-namespaces=demo-excluded
```

or we use the default mode and label all our namespaces with:
`goldilocks.fairwinds.com/enabled=true`

I think it's better that we use `on-by-default`.
I'm unsure if we think we need to disable any namespaces.

**Comment by NissesSenap** [comment]
I have excluded goldilocks from creating VPA in kube-system since we don't control the resources in there.
I have added a extra rbac rule so we can create VPA rules for cronjobs.
Ready for review, @jimgus @phillebaba

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## Unreleased
 
+### Added
+
+- [#506](https://github.com/XenitAB/terraform-modules/pull/506) Add VPA(Vertical Pod Autoscaling) as a module.
```
Comment (phillebaba) [inline CHANGELOG.md:11]: NIT: Add a space after VPA

File: modules/kubernetes/prometheus/templates/values.yaml.tpl [hunk 4]
```diff
@@ -31,11 +31,44 @@ kube-state-metrics:
...
+    - replicationcontrollers
+    - resourcequotas
+    - secrets
+    - services
+    - statefulsets
+    - storageclasses
+    - validatingwebhookconfigurations
+    - volumeattachments
+    - verticalpodautoscalers # not a default resource, see also: https://github.com/kubernetes/kube-state-metrics#enabling-verticalpodautoscalers
+  %{ endif }
```
Comment (phillebaba) [inline modules/kubernetes/prometheus/templates/values.yaml.tpl:65]: Seems stupid that we have to redefine all collectors but there does not seem to be another way :(

File: modules/kubernetes/vpa/templates/vpa-values.yaml.tpl [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  image:
+    # recommender.image.repository -- The location of the recommender image
+    repository: k8s.gcr.io/autoscaling/vpa-recommender
+    # recommender.image.pullPolicy -- The pull policy for the recommender image. Recommend not changing this
+    pullPolicy: Always
```
Comment (phillebaba) [inline modules/kubernetes/vpa/templates/vpa-values.yaml.tpl:18]: Why are we setting pull policy Always? We should really avoid using this.

**Review by phillebaba — APPROVED** [review]

**Pull request merged**